Documentation/cmdref/cilium-agent.md

```diff
@@ -42,6 +42,7 @@ cilium-agent [flags]
       --bpf-lb-mode string                                   BPF load balancing mode ("snat", "dsr", "hybrid") (default "snat")
       --bpf-lb-rss-ipv4-src-cidr string                      BPF load balancing RSS outer source IPv4 CIDR prefix for IPIP
       --bpf-lb-rss-ipv6-src-cidr string                      BPF load balancing RSS outer source IPv6 CIDR prefix for IPIP
+      --bpf-lb-wireguard                                     Enable Wireguard encryption for packets forwarded by Cilium LB (e.g. NodePort)
```

Comment:
Just a thought: Couldn't we pick a reasonable default without extra opt in/out? How is the encryption handled in case of IPsec for (kube-proxy) NodePort traffic where the backend is remote? Couldn't we default to the same behavior and document that? What if in the load-balancer there are some services where backends are outside of K8s scope, and thus have no context of wireguard? Probably it makes sense to stick to encryption by default for backends in the ipcache, and for everything external (aka world) to have no encryption. The east-west case with sock LB will presumably also stick to encryption, so north-south case could be consistent to that. Otoh, one could argue if we're just punting client packets from A to B they might have arrived at B in the first place, so what would be the point of encryption there, but it may be weird given these have the src IP of the cluster node which emit pkts in clear text.

Reply:
> Couldn't we pick a reasonable default without delegating extra opt in/out for users? 

Yeah, I was thinking about it, but my motivation for introducing the flag is the following:
1. Some users might not want to have the asymmetry (pod2host => cilium_wg0, host2pod=> eth0) introduced by the KPR encryption.
2. Not all users might need the KPR encryption.
3. Users who are running older than 5.8 kernel and has KPR enabled won't be able to enable Wireguard due to the absent `bpf_skb_change_head()` helper.

> How is the encryption handled in case of IPsec for (kube-proxy) NodePort traffic where the backend is remote?

Haven't tested, but I expect it to encrypt the traffic if the host2host encryption is enabled.

> What if in the load-balancer there are some services where backends are outside of K8s scope, and thus have no context of wireguard?

Such users would use `--bpf-lb-wireguard=false`.

> Probably it makes sense to stick to encryption by default for backends in the ipcache, and for everything external (aka world) to have no encryption.

Yep, but unfortunately we currently don't encrypt host2pod traffic (non-NodePort).

> The east-west case with sock LB will presumably also stick to encryption, so north-south case could be consistent to that.

Yep, that's the case today (except when a client is running host netns).

Reply:
Discussed offline. For now until we have host2pod encryption let's disable KPR encryption. /cc @gandro